kolena/_experimental/object_detection/utils.py

```diff
@@ -16,12 +16,120 @@
+            y_score_by_label[bbox_inf.label].append(bbox_inf.score)
+            y_true.append(0)
+            y_score.append(bbox_inf.score)
+    return y_true, y_score, dict(y_true_by_label), dict(y_score_by_label)
+
+
+def _compute_threshold_curves(
+    y_true: np.ndarray,
+    y_score: np.ndarray,
+    label: str,
+) -> Tuple[Curve, Curve]:
+    if len(y_score) >= 501:
+        thresholds = list(np.linspace(min(abs(y_score)), max(y_score), 501))[:-1]
+    else:
+        thresholds = np.unique(np.sort(y_score))
+        thresholds = thresholds[thresholds >= 0]
+        thresholds = thresholds.tolist()[:-1]
+
+    precisions: List[float] = []
+    recalls: List[float] = []
+    f1s: List[float] = []
+    for threshold in thresholds:
+        y_pred = [1 if score > threshold else 0 for score in y_score]
+        precision, recall, f1, _ = sklearn_metrics.precision_recall_fscore_support(
+            y_true,
+            y_pred,
+            average="binary",
+            zero_division=0,
+        )
+
+        precisions.append(precision)
+        recalls.append(recall)
+        f1s.append(f1)
+    return Curve(x=thresholds, y=f1s, label=label), Curve(x=recalls, y=precisions, label=label)
+
+
+def compute_pr_f1_plots(
+    all_matches: List[MulticlassInferenceMatches],
+    curve_label: str = "baseline",
+) -> List[CurvePlot]:
+    """
+    Creates a PR (precision and recall) curve and F1-threshold (confidence threshold) curve for the multiclass object
+    detection workflow. For `n` labels, each plot has `n+1` curves. One for the test case, and one per label.
+
+    :param all_matches: a list of multiclass or single class matching results.
+    :param curve_label: the label of the main curve.
+    :return: Two :class:`CurvePlot`s for the PR curves and F1-threshold curves for the test case and each label.
+    """
+    f1_curves: List[Curve] = []
+    pr_curves: List[Curve] = []
+
+    y_true, y_score, _, _ = _compute_sklearn_arrays(all_matches)
+    f1_curve, pr_curve = _compute_threshold_curves(np.array(y_true), np.array(y_score), curve_label)
+    f1_curves.append(f1_curve)
+    pr_curves.append(pr_curve)
+
+    # TODO: Uncomment when per class metrics are desired
+    # classes = sorted(y_true_by_label.keys())
+    # for label in classes:
+    #     y_true, y_score = y_true_by_label[label], y_score_by_label[label]
+    #     if len(y_true) > 0:
+    #         f1_curve, pr_curve = _compute_threshold_curves(np.array(y_true), np.array(y_score), label)
+    #         f1_curves.append(f1_curve)
+    #         pr_curves.append(pr_curve)
+
+    threshold_f1 = CurvePlot(
+        title="F1-Score vs. Confidence Threshold",
+        x_label="Confidence Threshold",
+        y_label="F1-Score",
+        curves=f1_curves,
+    )
+
+    pr = CurvePlot(
+        title="Precision vs. Recall",
+        x_label="Recall",
+        y_label="Precision",
+        curves=pr_curves,
+    )
+
+    return [threshold_f1, pr]
```

Comment:
It's a weird public API to have the creation of the F1-vs-T plot and the PR curve coupled like this. Is there a way to keep the perf benefits of coupling them without jamming both into a single external-facing function?